### ISSUE TITLE
Bug: 6.13.0 breaks the `deprecation` and `etc` ESLint plugins

### ISSUE THREAD

**Issue by boris-petrov** [typescript-eslint/typescript-eslint#8002]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Issue Description

Check the reproduction. Running it leads to:

```
/home/boris/code/eslint-typescript-bug/foo.ts
  1:40  error  Unused eslint-disable directive (no problems were reported from 'deprecation/deprecation')
```

Downgrading ESLint TypeScript to 6.12.0 fixes the problem - i.e. this directive is not marked as unused (which is the correct behavior).\

The same issue also appears with the `etc/no-internal` rule (which is not showcased here).

### Reproduction Repository Link

https://github.com/boris-petrov/eslint-typescript-bug

### Repro Steps

1) clone the repo
2) `pnpm install`
3) `./node_modules/.bin/eslint --cache --max-warnings=0 --report-unused-disable-directives .`

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `6.13.0` |
| `@typescript-eslint/parser`            | `6.13.0` |
| `@typescript-eslint/scope-manager`     | `6.13.0` |
| `@typescript-eslint/typescript-estree` | `6.13.0` |
| `@typescript-eslint/type-utils`        | `6.13.0` |
| `@typescript-eslint/utils`             | `6.13.0` |
| `TypeScript`                           | `5.3.2` |
| `ESLint`                               | `8.54.0` |
| `node`                                 | `21.2.0` |

**Comment by bradzacher** [comment]
https://github.com/typescript-eslint/typescript-eslint/pull/7999

**Comment by bradzacher** [comment]
The fix has been released https://github.com/typescript-eslint/typescript-eslint/releases/tag/v6.13.1
